src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -896,9 +896,11 @@ where
             if as_of.iter().all(|t| !upper.less_equal(t)) {
                 let committed_upper = frontiers[1].frontier();
                 if as_of.iter().all(|t| !committed_upper.less_equal(t)) {
-                    // Compute the frontier of the binding that is *less than* the committed upper.
+                    // We accumulate all the bindings that are not beyond the committed upper,
+                    // which has the effect of calculating the value of the previous binding than
+                    // the one directly at committed upper.
                     let idx =
-                        ready_times.partition_point(|(_, t, _)| !committed_upper.less_than(t));
+                        ready_times.partition_point(|(_, t, _)| !committed_upper.less_equal(t));
```

Comment:
can we add some justification as to why we want the previous binding, as opposed to the one _at_ the upper we are trying to reclock? to me, its unclear

Reply:
done